# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/syscall6.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

pselect needs all 6 possible arguments, ...

  - Rene Rebe <rene@exactcode.de>

--- dietlibc-0.32/i386/unified.S.orig	2004-01-15 17:35:43.000000000 +0000
+++ dietlibc-0.32/i386/unified.S	2011-01-09 21:55:35.051457158 +0000
@@ -30,6 +30,7 @@
 .size _exit,__unified_syscall-_exit
 	movzbl	%al, %eax
 .L1:
+	push	%ebp
 	push	%edi
 	push	%esi
 	push	%ebx
@@ -37,11 +38,12 @@
 	/* we use movl instead of pop because otherwise a signal would
 	   destroy the stack frame and crash the program, although it
 	   would save a few bytes. */
-	movl	0x10(%edi),%ebx
-	movl	0x14(%edi),%ecx
-	movl	0x18(%edi),%edx
-	movl	0x1c(%edi),%esi
-	movl	0x20(%edi),%edi
+	movl	0x14(%edi),%ebx
+	movl	0x18(%edi),%ecx
+	movl	0x1c(%edi),%edx
+	movl	0x20(%edi),%esi
+	movl	0x28(%edi),%ebp
+	movl	0x24(%edi),%edi
 #ifdef WANT_SYSENTER
 	call	*.Lvsyscall /* 0xffffe000 */
 #else
@@ -63,6 +65,7 @@
 	pop	%ebx
 	pop	%esi
 	pop	%edi
+	pop	%ebp
 
 /* here we go and "reuse" the return for weak-void functions */
 #include "dietuglyweaks.h"
